core/modules/zmq_mdlw/include/klepsydra/zmq_core/zhelpers.hpp

```diff
@@ -73,31 +73,90 @@
 #endif
 
 //  Provide random number from 0..(num-1)
-#define within(num) (int) ((float) (num) * random () / (RAND_MAX + 1.0))
+#define within(num) (int) ((float)((num) * random ()) / (RAND_MAX + 1.0))
+
+//  Receive 0MQ string from socket and convert into C string
+//  Caller must free returned string.
+inline static char *
+s_recv(void *socket, int flags = 0) {
+	zmq_msg_t message;
+	zmq_msg_init(&message);
+
+	int rc = zmq_msg_recv(&message, socket, flags);
+
+	if (rc < 0)
+		return nullptr;           //  Context terminated, exit
+
+	size_t size = zmq_msg_size(&message);
+	char *string = (char*)malloc(size + 1);
+	memcpy(string, zmq_msg_data(&message), size);
+	zmq_msg_close(&message);
+	string[size] = 0;
+	return (string);
+}
 
 //  Receive 0MQ string from socket and convert into string
-static std::string
-s_recv (zmq::socket_t & socket) {
+inline static std::string
+s_recv (zmq::socket_t & socket, int flags = 0) {
 
     zmq::message_t message;
-    socket.recv(&message);
+    socket.recv(&message, flags);
 
     return std::string(static_cast<char*>(message.data()), message.size());
 }
 
+inline static bool s_recv(zmq::socket_t & socket, std::string & ostring, int flags = 0)
+{
+	zmq::message_t message;
+	bool rc = socket.recv(&message, flags);
+
+	if (rc) {
+		ostring = std::string(static_cast<char*>(message.data()), message.size());
+	}
+	
+	return (rc);
+}
+
+//  Convert C string to 0MQ string and send to socket
+inline static int
+s_send(void *socket, const char *string, int flags = 0) {
+	int rc;
+	zmq_msg_t message;
+	zmq_msg_init_size(&message, strlen(string));
+	memcpy(zmq_msg_data(&message), string, strlen(string));
+	rc = zmq_msg_send(&message, socket, flags);
+	assert(-1 != rc);
+	zmq_msg_close(&message);
+	return (rc);
+}
+
 //  Convert string to 0MQ string and send to socket
-static bool
-s_send (zmq::socket_t & socket, const std::string & string) {
+inline static bool
+s_send (zmq::socket_t & socket, const std::string & string, int flags = 0) {
 
     zmq::message_t message(string.size());
     memcpy (message.data(), string.data(), string.size());
 
-    bool rc = socket.send (message);
+    bool rc = socket.send (message, flags);
     return (rc);
 }
 
 //  Sends string as 0MQ string, as multipart non-terminal
-static bool
+inline static int
+s_sendmore(void *socket, char *string) {
+	int rc;
+	zmq_msg_t message;
+	zmq_msg_init_size(&message, strlen(string));
```

Comment:
Same - strnlen instead.

Reply:
cannot do it, I am afraid. This file comes from ZMQ and it is used by cppzmq. Gonna keep it that way for now....